Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaDayViewModel.kt

```diff
@@ -64,9 +64,8 @@ class AgendaDayViewModel(
     }
 
     fun fetchScheduleData() {
-        if (scheduleRows.value == null) {
-            firebaseHelper.eventDatabase.addValueEventListener(dataListener)
-        }
+        firebaseHelper.eventDatabase.removeEventListener(dataListener)
+        firebaseHelper.eventDatabase.addValueEventListener(dataListener)
```

Comment:
this looks harmless enough, but i'm curious why it's necessary to remove and re-add each time 🤔

Reply:
fair question...was defensive coding from the if statement I took out...not sure why the if was needed.
I should look at the firebase listener handling code...not sure if it's a map or set underneath.  If it is, the remove isn't needed.  If it's a list/array, it is needed so you don't get dup notifications.

Reply:
Dug through the decompiled Firebase code and it does look like it's implemented as a list:
private final List<EventRegistration> eventRegistrations;

So it is safer to leave the remove in....